src/collaboration-service/controllers/session-controller.js

```diff
@@ -118,3 +118,18 @@ exports.editSession = async (req, res) => {
   }
 }
 
+exports.findSessionWithUid = async (req, res) => {
+  try {
+    const uid = req.params.uid;
+    const session = await Session.findOne({
+      $or: [
+        { firstUserId: uid },
+        { secondUserId: uid }
+      ]
+    });
+    return res.status(200).json(session);
+  } catch (err) {
+    console.log(err);
+    return res.status(500).json({ message: "Error fetching session" });
```

Comment:
I think returning an error when the session does not exist is not good? There might be other cases to handle.

Maybe can use aggregate instead?

Reply:
The Mongoose findOne returns null if no session found, so this is fine if the FE handles the null case

Reply:
> I think returning an error when the session does not exist is not good? There might be other cases to handle.
> 
> Maybe can use aggregate instead?

When no session, theres no error, jus that session is null.
so on front end it checks "if session" to handle null case.

Reply:
Should be okay then